records_mover/db/db_conn_mixin.py

```diff
@@ -0,0 +1,38 @@
+import logging
```

Comment:
i like this mixin. I feel like somewhere in the code this PR adds we should add a comment explaining our general constraints of wanting to not break existing python code that might have relied on this "db" property. Which greatly shaped the structure and changes of the code you wrote. I imagine someone reading this code in the future without realizing this constraint would be very confused about why some things are the way they are now.

Maybe somewhere in this class is the right place to add such a comment? Maybe near `check_db_conn_engine`? not sure

Reply:
I think that's a great idea, will get on this

Reply:
Added an explainer in `check_db_conn_engine`